docs/proposals/kuratorcd/kurator-cd.md

```diff
@@ -0,0 +1,563 @@
+
+Unlike Flagger, we may need to adjust Unified Continuous Delivery to reflect our new strategy and decisions.
+
+#### Unified Continuous Delivery API
+
+Kurator is designed to unify the installation of Flagger as a fleet plugin in a given single or multiple clusters.
+
+Then use the Kurator application to distribute the Flagger configuration. Kurator's unified configuration distribution.
+
+Kurator puts the Continuous Delivery's api under the [Application](https://github.com/kurator-dev/kurator/blob/main/pkg/apis/apps/v1alpha1/types.go) CRD, so that when Kurator deploys the workload in the target cluster, it also deploys the corresponding Continuous Delivery policy.
+
+Here's the preliminary design for the Unified Continuous Delivery:
+
+```console
+// ApplicationSyncPolicy defines the configuration to sync an artifact.
+// Only oneof `kustomization` or `helm` can be specified to manage application sync.
+// ApplicationSyncPolicy distributes the Continuous Delivery configuration 
+// at the same time as the application deployment, if needed. 
+type ApplicationSyncPolicy struct {
+    // Name defines the name of the sync policy.
+    // If unspecified, a name of format `<application name>-<index>` will be generated.
+    // +optional
+    Name string `json:"name,omitempty"`
+
+    // Kustomization defines the configuration to calculate the desired state
+    // from a source using kustomize.
+    // +optional
+    Kustomization *Kustomization `json:"kustomization,omitempty"`
+    // HelmRelease defines the desired state of a Helm release.
+    // +optional
+    Helm *HelmRelease `json:"helm,omitempty"`
+
+    // Destination defines the destination for the artifact.
+    // If specified, it will override the destination specified at Application level.
+    // +optional
+    Destination *ApplicationDestination `json:"destination"`
+
+    // Rollout defines the rollout Configurations to be used.
+    // If specified, a uniform Continuous Delivery policy is configured for this installed object.
+    // +optional
+    Rollout *RolloutConfig `json:"rolloutPolicy,omitempty"`
+}
+
+type RolloutConfig struct {
+    // Testload defines Whether to install testload for users. Default is true.
+    // testload generates traffic during canary analysis.
+    // If set it to false, user need to install the testload himself.
+    // If set it to true or leave it blank, Kurator will install the flagger's testload.
+    // +optional
+    TestLoad bool `json:"testLoad,omitempty"`
+
+    // Kurator only supports istio for now.
+    // New Provider will be added later.
+    // +optional
+    TrafficRoutingProvider string `json:"trafficRoutingProvider,omitempty"`
+
+    // Workload specifies what workload to deploy the test to. 
+    // Workload of type deployment or daemonSet.
+    Workload *WorkloadReference `json:"workload"`
+
+    // Port of the Workload which want to test.
+    Port int32 `json:"port"`
```

Comment:
confused about this.

So if there are two ports defined in the service, user can specify one here?